source/lambda/kendraDataSourceCreator/lambda_function.py

```diff
@@ -0,0 +1,41 @@
+import boto3
+import os
+
+def on_create(event, context):
+    print("creating an s3 data source and connector for the Covid dataset")
+    kendra_client = boto3.client('kendra')
+    kendraIndexId = os.environ['KENDRA_INDEX_ID']
+    print(kendraIndexId)
+    dataBucketName = os.environ['DATA_BUCKET_NAME']
+    roleArn = os.environ['KENDRA_ROLE_ARN']
+
+    create_data_source_response = kendra_client.create_data_source(
+        Name='CovidDataset',
+        IndexId= kendraIndexId,
+        Type='S3',
+        Configuration={
+            'S3Configuration': {
+                'BucketName': dataBucketName,
+            }
+        },
+        Description='Covid pdfs data',
+        RoleArn=roleArn
+    )
+    dataSourceId = create_data_source_response['Id']
+    print(create_data_source_response)
+    print("synching the s3 data source")
+    response = kendra_client.start_data_source_sync_job(
+        Id=dataSourceId,
+        IndexId=kendraIndexId
+    )
+    print(response)
+    return {'PhysicalResourceId' : dataSourceId}
+   
+def on_update(event, context):
+    print("Cannot update KMS key for Kendra Index. Please delete the index and try again.")
```

Comment:
Is this the only case for update for KMS key id? You may want to check other updates also if possible such as index id or role arn etc
Also depending on the update, you may also have to throw exceptions. Please test this out with the different possibilities for updates

Reply:
I thought updating the stack using yarn deploy will create all resources newly with new uuids at the end. We cannot update the kendra index as the kms key id cannot be changed for the index hence I thought this comment was appropriate.

Reply:
Updating the stack does not change uuid.
Also if there are changes made to resources like KMS ID, dont you need to throw and exception?